pkg/client/events/replay_client.go

```diff
@@ -117,7 +118,16 @@ func NewEventsReplayClient[T any](
 	}
 
 	// Concurrently publish events to the observable emitted by replayObsCache.
-	go rClient.goPublishEvents(ctx)
+	go func() {
```

Comment:
How confident are you that this is the correct approach?

Going to delegate to @bryanchriswhite who has more context, but reply client was designed and intended to use `publishEvents` directly.

Reply:
I don't believe it is, the correct approach is with returning an error channel of some sort so SDK consumer can handle it. This was a compromise for short term solution with minimal changes.